packages/telegram-cloud-storage-stamper/src/index.ts

```diff
@@ -0,0 +1,296 @@
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    // check to see that the stamper was initialized
+    if (!this.stamper) {
+      throw new TelegramCloudStorageStamperError(
+        "Cannot stamp with uninitialized telegram stamper, try running .create() or .setSigningKey()"
+      );
+    }
+
+    return await this.stamper.stamp(payload);
+  }
+
+  // set the api key that is used by the telegram cloud storage for signing requests
+  async setSigningKey(config?: TTelegramCloudStorageStamperConfig) {
+    if(!config) {
+      // try to get api key at default location and set the stamper to use that
+      // passing the DEFAULT_TURNKEY_CLOUD_STORAGE_KEY is not necessary since getAPIKey will use that by default
+      let { apiPublicKey, apiPrivateKey } = await this.getAPIKey(DEFAULT_TURNKEY_CLOUD_STORAGE_KEY);
+
+      if(apiPublicKey && apiPrivateKey) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey,
+          apiPrivateKey
+        })
+        return true
+      }
+    } else if(config.cloudStorageKey && !config.cloudStorageAPIKey) {
+      // try to get api key at specified location and set the stamper to use that
+      let { apiPublicKey, apiPrivateKey } = await this.getAPIKey(config.cloudStorageKey);
+
+      if(apiPublicKey && apiPrivateKey) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey,
+          apiPrivateKey
+        })
+        return true
+      }
+    } else if(!config.cloudStorageKey && config.cloudStorageAPIKey) {
+      // try to set api key at default location and set the stamper to use that
+      // passing the DEFAULT_TURNKEY_CLOUD_STORAGE_KEY is not necessary since insertAPIKey will use that by default
+      let stored = await this.insertAPIKey(config.cloudStorageAPIKey.apiPublicKey, config.cloudStorageAPIKey.apiPrivateKey, DEFAULT_TURNKEY_CLOUD_STORAGE_KEY);
+
+      if(stored === true) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey: config.cloudStorageAPIKey.apiPublicKey,
+          apiPrivateKey: config.cloudStorageAPIKey.apiPrivateKey
+        })
+        return true
+      }
+    } else if(config.cloudStorageKey && config.cloudStorageAPIKey) {
+      // try to set api key at specified location and set the stamper to use that
+      let stored = await this.insertAPIKey(config.cloudStorageAPIKey.apiPublicKey, config.cloudStorageAPIKey.apiPrivateKey, config.cloudStorageKey);
+
+      if(stored === true) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey: config.cloudStorageAPIKey.apiPublicKey,
+          apiPrivateKey: config.cloudStorageAPIKey.apiPrivateKey
+        })
+        return true
+      }
+    }
+    return false
+  }
+
+
+  async insertItem(key: string, value: string) {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    try {
+      return await this.setCloudStorageItem(key, value)
+    } catch {
+      return false
+    }
```

Comment:
I'm not sure this is a good pattern in general. I'd rather let the exception bubble up to the caller so they can handle it the way they see fit. With "false" returned it's too easy to forget to check this value.

More generally this "check return value" pattern is very C-like or Go-like; most JS devs expect exceptions to be thrown when something goes wrong rather than boolean values indicating success.

Reply:
There are really 2 sets of functions available for the developer to use. The lower level `setCloudStorageItem`, or `setItem`. I tried to simplify them with the `setItem` variants of the functions and still have setCloudStorageItem available to throw exceptions if the developers want to use them

Reply:
They will always be `TelegramCloudStorageStamperError`'s You can see [here](https://github.com/tkhq/sdk/blob/a375c266e55596bea09a2f493dcc3ee2af2e34bc/packages/telegram-cloud-storage-stamper/src/index.ts#L250-L256). There are some lower level functions that will throw errors and then some higher level functions which I thought would be easier for developers to use. Ill make a distinction about this in the README so that is clear.

Reply:
This logic has been overhauled and is no longer relevant